apis/slo/v1alpha1/nodeslo_types.go

```diff
@@ -23,6 +23,107 @@ import (
+	// stops when usage is below `memory.wmark_low`. Basically,
+	// `memory.wmark_high` := min(memory.high, memory.limit_in_bytes) * memory.wmark_scale_factor
+	// `memory.wmark_low` := min(memory.high, memory.limit_in_bytes) * (memory.wmark_ratio - memory.wmark_scale_factor)
+	// WmarkRatio specifies `memory.wmark_ratio` that help calculate `memory.wmark_high`, which triggers async
+	// memory reclamation when memory usage exceeds.
+	// Close: 0. Recommended: 95.
+	WmarkRatio *int64 `json:"wmarkRatio,omitempty"`
+	// WmarkScalePermill specifies `memory.wmark_scale_factor` that helps calculate `memory.wmark_low`, which
+	// stops async memory reclamation when memory usage belows.
+	// Close: 50. Recommended: 20.
+	WmarkScalePermill *int64 `json:"wmarkScalePermill,omitempty"`
+
+	// wmark_min_adj (Anolis OS required)
+	// WmarkMinAdj specifies `memory.wmark_min_adj` which adjusts per-memcg threshold for global memory
+	// reclamation. Lower the factor brings later reclamation.
+	// The adjustment uses different formula for different value range.
+	// [-25, 0)：global_wmark_min' = global_wmark_min + (global_wmark_min - 0) * wmarkMinAdj
+	// (0, 50]：global_wmark_min' = global_wmark_min + (global_wmark_low - global_wmark_min) * wmarkMinAdj
+	// Close: [LSR:0, LS:0, BE:0]. Recommended: [LSR:-25, LS:-25, BE:50].
+	WmarkMinAdj *int64 `json:"wmarkMinAdj,omitempty"`
+
+	// TODO: enhance the usages of oom priority and oom kill group
+	PriorityEnable *int64 `json:"priorityEnable,omitempty"`
+	Priority       *int64 `json:"priority,omitempty"`
+	OomKillGroup   *int64 `json:"oomKillGroup,omitempty"`
+}
+
+type PodMemoryQOSPolicy string
+
+const (
+	// PodMemoryQOSPolicyDefault indicates pod inherits node-level config
+	PodMemoryQOSPolicyDefault PodMemoryQOSPolicy = "default"
+	// PodMemoryQOSPolicyNone indicates pod disables memory qos
+	PodMemoryQOSPolicyNone PodMemoryQOSPolicy = "none"
+	// PodMemoryQOSPolicyAuto indicates pod uses a recommended config
+	PodMemoryQOSPolicyAuto PodMemoryQOSPolicy = "auto"
+)
+
+type PodMemoryQOSConfig struct {
+	// Policy indicates the qos plan; use "default" if empty
+	Policy    PodMemoryQOSPolicy `json:"policy,omitempty"`
+	MemoryQOS `json:",inline"`
+}
+
+// MemoryQOSCfg stores node-level config of memory qos
+type MemoryQOSCfg struct {
+	// Enable indicates whether the memory qos is enabled (Cluster default: false).
+	// This field is used for node-level control, while pod-level configuration is done with MemoryQOS and `Policy`
+	// instead of an `Enable` option. Please view the differences between MemoryQOSCfg and PodMemoryQOSConfig structs.
+	Enable    *bool `json:"enable,omitempty"`
+	MemoryQOS `json:",inline"`
+}
+
+type ResourceQOS struct {
+	MemoryQOS *MemoryQOSCfg `json:"memoryQOS,omitempty"`
+}
+
+type ResourceQOSStrategy struct {
+	// ResourceQOS for LSR pods.
+	LSRClass *ResourceQOS `json:"lsrClass,omitempty"`
```

Comment:
The class suffix is not required. Consider defining ResourceQOSStrategy as follows:
```go
type ResourceQOSStrategy struct {
        Type string `json:"type"`
        ResourceQoS *ResourceQoS `json:"resourceQoS,omitempty"`
}
```

Reply:
we only support resource configs for stable QoS class, i.e. SYSTEM, LSR, LS, BE.